src/Shouldly/Internals/EqualityComparer.cs

```diff
@@ -91,6 +92,45 @@ public bool Equals(T x, T y)
             return object.Equals(x, y);
         }
 
+        private static bool TryGetEnumerable(object obj, out IEnumerable enumerable)
+        {
+            enumerable = obj as IEnumerable;
+
+            if (enumerable == null && obj != null)
+            {
+                var type = obj.GetType();
+                if (type.IsMemory(out var elementType))
+                {
+                    var readOnlyMemory = type.GetMethods(BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly)
+                        .SingleOrDefault(method =>
+                            method.Name == "op_Implicit"
+                            && method.GetParameters()[0].ParameterType == type
+                            && method.ReturnType.IsReadOnlyMemory(out var returnElementType)
+                            && returnElementType == elementType)
+                        ?.Invoke(null, new[] { obj });
+
+                    if (readOnlyMemory != null)
+                    {
+                        enumerable = ToEnumerable(readOnlyMemory, elementType);
+                    }
+                }
+                else if (type.IsReadOnlyMemory(out elementType))
+                {
+                    enumerable = ToEnumerable(obj, elementType);
+                }
+            }
+
+            return enumerable != null;
+        }
+
+        private static IEnumerable ToEnumerable(object readOnlyMemory, Type elementType)
+        {
+            return (IEnumerable)Type.GetType("System.Runtime.InteropServices.MemoryMarshal, System.Memory")
+                ?.GetMethod("ToEnumerable", BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly)
```

Comment:
It's best to specify parameter types so that this doesn't crash if an overload is added.

Reply:
I wanted to use this overload of `GetMethod()` but it only support .NET Standard 2.0 and above - https://docs.microsoft.com/en-us/dotnet/api/system.type.getmethod?view=netframework-4.8#System_Type_GetMethod_System_String_System_Reflection_BindingFlags_System_Reflection_Binder_System_Type___System_Reflection_ParameterModifier___

Below is the modified code which fails to compile.
```
            return (IEnumerable)Type.GetType("System.Runtime.InteropServices.MemoryMarshal, System.Memory")
                ?.GetMethod("ToEnumerable", BindingFlags.Public | BindingFlags.Static | BindingFlags.DeclaredOnly, null, new[] { typeof(ReadOnlyMemory<T>) }, null)
                ?.MakeGenericMethod(elementType)
                .Invoke(null, new[] { readOnlyMemory });

```

Reply:
![image](https://user-images.githubusercontent.com/4055444/77838939-49912880-71c4-11ea-8518-7cceee790991.png)

Reply:
Ah, that's too bad. We could cross our fingers and hope that an overload of MemoryMarshal.ToEnumerable is never added, or we could use GetMethods+SingleOrDefault like you did above, or we could replace netstandard1.3 with netstandard2.0.

@JosephWoodward What are your thoughts?

Reply:
GetMethods+SingleOrDefault sounds like a small enough change as opposed to upgrading to netstandard2.0

The library may stop supporting older .NET versions which may have unintended consequences.